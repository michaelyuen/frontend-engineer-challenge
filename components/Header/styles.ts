import styled from "styled-components";

export const HeaderContainer = styled.header`
  display: flex;
  flex-direction: column;
  padding: 1em;
  width: 100%;

  div {
    background-image: var(--logo);
    background-size: contain;
    background-repeat: no-repeat;
    height: 24px;
    width: 180px;
  }

  p {
    font-size: 0.8em;
    margin-top: 0.5em;
    margin-bottom: 0;
  }
`;
