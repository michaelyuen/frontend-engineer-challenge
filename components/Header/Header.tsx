import { HeaderContainer } from "./styles";

export const Header = () => {
  return (
    <HeaderContainer>
      {/* TODO(my): Replace with Image, switching logo src based on theme */}
      <div aria-label="Koala logo" />
      <p>
        <span>FE Coding Challenge</span>
        <span>Frontend Engineer Coding Challenge</span>
      </p>
    </HeaderContainer>
  );
};
