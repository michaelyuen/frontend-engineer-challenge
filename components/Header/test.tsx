import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { Header } from "./Header";

describe("Header", () => {
  test("renders", () => {
    render(<Header />);
    expect(screen.getByLabelText("Koala logo")).toBeInTheDocument();
    expect(
      screen.getByText("Frontend Engineer Coding Challenge")
    ).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(<Header />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
