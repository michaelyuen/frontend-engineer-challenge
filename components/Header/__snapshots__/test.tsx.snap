// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    class="sc-bcXHqe bcxMoQ"
  >
    <div
      aria-label="Koala logo"
      role="img"
    />
    <p>
      <span>
        FE Coding Challenge
      </span>
      <span>
        Frontend Engineer Coding Challenge
      </span>
    </p>
  </header>
</div>
`;
